fdbserver/workloads/SpecialKeySpaceCorrectness.actor.cpp

```diff
@@ -0,0 +1,204 @@
+/*
+ * SpecialKeySpaceCorrectness.actor.cpp
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2020 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "fdbclient/NativeAPI.actor.h"
+#include "fdbclient/ReadYourWrites.h"
+#include "fdbclient/SpecialKeySpace.actor.h"
+#include "fdbserver/TesterInterface.actor.h"
+#include "fdbserver/workloads/workloads.actor.h"
+#include "flow/actorcompiler.h"
+
+class SKSCTestImpl : public SpecialKeyRangeBaseImpl {
+public:
+	explicit SKSCTestImpl(KeyRef start, KeyRef end) : SpecialKeyRangeBaseImpl(start, end) {}
+	virtual Future<Standalone<RangeResultRef>> getRange(Reference<ReadYourWritesTransaction> ryw,
+	                                                    KeyRangeRef kr) const {
+		ASSERT(range.contains(kr));
+		auto resultFuture = ryw->getRange(kr, CLIENT_KNOBS->TOO_MANY);
```

Comment:
What if there are more results to read?
typically, we assert that there is no more keys to read in the result so that the error is easier to find.

Reply:
Add an assertion to make sure we get all results. Thanks.